src/states/node.rs

```diff
@@ -1633,15 +1638,27 @@ impl Node {
         };
 
         if self.is_proper() && !force_via_proxy {
-            let targets = self.peer_mgr
+            let targets: HashSet<_> = self.peer_mgr
                 .routing_table()
-                .targets(&routing_msg.dst.to_destination(), exclude, route as usize)?;
-            Ok(self.peer_mgr.get_peer_ids(&targets))
+                .targets(&routing_msg.dst.to_destination(), *hop, route as usize)?
+                .into_iter()
+                .filter(|target| !sent_to.contains(target))
+                .collect();
+            let new_sent_to =
+                if self.peer_mgr.routing_table().is_in_our_group(routing_msg.dst.name()) {
```

Comment:
The target could be the name of a data chunk or something else, other than the name of a node in our group.
This should probably be `our_group_prefix().matches(routing_msg.dst.name())`.